### PR TITLE
[DOCS] Relocate `node` module content

### LINKED ISSUE #53307: [DOCS] Move node topics out of modules
As part of the Modules reorg to improve discovery and navigation, find a better home for the following topics:

* [x] Node #56376
* [x] Thread pool #55814

### PR DESCRIPTION
**Pull request by jrodewig**
Moves `node` content from the [Modules][0] section to the
[Configuring Elasticsearch][1] section.

Supporting changes:
* Retitles page `Node settings`
* Increments several headings
* Removes several unneeded `[float]` attributes
* Replaces remaining `[float]` attributes with `[discrete]`
* Updates the `Other node settings` section

Closes #53307

[0]: https://www.elastic.co/guide/en/elasticsearch/reference/master/modules.html
[1]: https://www.elastic.co/guide/en/elasticsearch/reference/master/settings.html

### CODE REVIEW

File: docs/reference/modules/node.asciidoc [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [[modules-node]]
-== Node
+=== Node settings
```
Comment (debadair) [inline docs/reference/modules/node.asciidoc:2]: This topic is our "source of truth" for node definitions and is referenced in customer agreements. I agree we need a "node settings" topic, but having our official node info under "node settings" is confusing. As a follow-on, we need to rework this to split apart the settings from the definitions. IOW, merging to master is fine, but the content needs to be reworked before it becomes "current".

**Review by debadair — APPROVED** [review]
LGTM

File: docs/reference/modules/node.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/modules/node.asciidoc]: I'm good with leaving this as `node`. I'm not actually sure it makes sense to split the definitions from the settings here as they're pretty connected.

Reverted with b008abb4a11

**Comment by jrodewig** [comment]
### Backport commits
master fabb3e1
7.x 4dd05cc
7.8 7a6f7d7 
7.7 0cd532b25359b456ff7f74fc3261e5e7b5fb72a1

**Pull request merged**